e2e/docker_test.go

```diff
@@ -175,28 +191,47 @@ func setupPathDocker(
 		}()
 	}
 
-	// Handle termination signals
-	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+	// Create a context that we can cancel
+	ctx, cancel := context.WithCancel(context.Background())
+
+	// Set up signal handling
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, os.Interrupt, syscall.SIGTERM)
+
+	// Create a channel to wait for cleanup completion
+	cleanupDone := make(chan struct{})
+
+	// Handle signals
 	go func() {
-		for sig := range c {
-			fmt.Printf("exit signal %d received\n", sig)
+		select {
+		case <-signalChan:
+			fmt.Println("\nReceived Ctrl+C, cleaning up containers...")
```

Comment:
This does not appear to work. When I hit Ctrl+C with the test running it hangs on this log:
<img width="651" alt="Screenshot 2025-04-25 at 16 29 36" src="https://github.com/user-attachments/assets/24aee1f7-e03e-4d69-8eba-a0afbce2def7" />